src/water_cmd.cpp

```diff
@@ -452,10 +454,22 @@ CommandCost CmdBuildCanal(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 	TileArea ta(tile, p1);
 
 	/* Outside the editor you can only drag canals, and not areas */
-	if (_game_mode != GM_EDITOR && ta.w != 1 && ta.h != 1) return CMD_ERROR;
+	if (_game_mode != GM_EDITOR) {
+		TileArea ta(tile, p1);
```

Comment:
The exact same line `TileArea ta(tile, p1);` is already present above. In fact, that whole chunk seems unnecessary.

Reply:
Oops, thanks. I missed [the solo TileArea being deleted](https://github.com/JGRennison/OpenTTD-patches/commit/72b72dbec37dc6cc8a26a4c093613e7857eeb3b5#diff-2435a884057fcf692871728ce3d0726330c4120b00d7b07514eaa0ec310b0fd5L402) in JGR's original commit.